### ISSUE TITLE
cmake add_subdirectory

### ISSUE THREAD

**Issue by RobotCaleb** [m-mizutani/libfluent#3]
I'm trying to use libfluent in our own project. Lines 51 and 59 of your CMakeLists.txt cause issues with this.
We are including `msgpack-c` and `libfluent` using `add_subdirectory` calls. We have no desire to build libfluent separately and install.

Line 51 causes:
`CMake Error: install(EXPORT "fluentConfig" ...) includes target "fluent-shared" which requires target "msgpackc" that is not in the export set.`

If line 51 is commented out, line 59 causes:
`CMake Error: INSTALL(EXPORT) given unknown export "fluentConfig"`

If I comment both of them out then everything works quite well.
I'm not familiar enough with cmake to know a proper workaround for this, but as we intend to use this library as a submodule and contribute any changes upstream if we have any, I'd like to not have to maintain a custom version.

Do you have any input regarding my issue?

**Comment by m-mizutani** [comment]
Thank you for reporting. Let me reproduce the issue.

**Comment by RobotCaleb** [comment]
@m-mizutani : I threw this together to help you out.

```mkdir test
cd test
mkdir build
git init
git submodule add https://github.com/m-mizutani/libfluent.git
git submodule add https://github.com/msgpack/msgpack-c.git
echo "cmake_minimum_required(VERSION 2.6)" > CMakeLists.txt
echo "add_subdirectory(msgpack-c)" >> CMakeLists.txt
echo "add_subdirectory(libfluent)" >> CMakeLists.txt
cd build
cmake ../
```

**Comment by m-mizutani** [comment]
Sorry for late reply. I was investigating EXPORT mechanism of cmake because I'm also not an expert of cmake.

I found two way to avoid the error.

1. use `-DFLUENT_INSTALL=OFF` option that is implemented at b75a8b8 when running cmake command with your CMakeLists.txt. The option makes installation commands including export cmake.config function disable. Then, your cmake command will not fail on `install(EXPORT "fluentConfig" ...)`
2. use `ExternalProject_Add` command in your CMakeLIsts.txt. I thought it's a proper way if you allow to install msgpack and libfluent into your development environment. Please see details of ExternalProject_Add from folloing URL.

https://coderwall.com/p/y3zzbq/use-cmake-enabled-libraries-in-your-cmake-project

Let me close the issue, but please feel free to reopen if you have more issue.
